ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/Run.hs

```diff
@@ -257,85 +278,83 @@ mkLoEVar SchedulerConfig {scEnableLoE}
   | otherwise
   = pure LoEDisabled
 
--- | Construct STM resources, set up ChainSync and BlockFetch threads, and
--- send all ticks in a 'PointSchedule' to all given peers in turn.
-runPointSchedule ::
+mkStateTracer ::
+  IOLike m =>
+  SchedulerConfig ->
+  GenesisTest TestBlock s ->
+  PeerSimulatorResources m TestBlock ->
+  ChainDB m TestBlock ->
+  m (Tracer m ())
+mkStateTracer schedulerConfig GenesisTest {gtBlockTree} PeerSimulatorResources {psrHandles, psrPeers} chainDb
+  | scTraceState schedulerConfig
+  , let getCandidates = viewChainSyncState psrHandles CSClient.csCandidate
+        getCurrentChain = ChainDB.getCurrentChain chainDb
+        getPoints = traverse readTVar (srCurrentState . prShared <$> psrPeers)
+  = peerSimStateDiagramSTMTracerDebug gtBlockTree getCurrentChain getCandidates getPoints
+  | otherwise
+  = pure nullTracer
```

Comment:
Shouldn't trace-related functions go to `Trace.hs`?

Reply:
yes, but it would cause so many dependency cycles that the remaining code would almost be identical